homeassistant/components/wled/light.py

```diff
@@ -383,6 +406,25 @@ async def async_effect(
 
         await self.coordinator.wled.segment(**data)
 
+    @wled_exception_handler
+    async def async_random_effect(self) -> None:
+        """Activate random effect for a WLED light."""
+        data = {ATTR_SEGMENT_ID: self._segment}
+        effect = data[ATTR_EFFECT] = random.choice(self.effect_list)
```

Comment:
This is adding artificial features to the WLED integration. This is not something we should add to the integration.

Reply:
Any suggestions how this could be improved to make it acceptable? Exposing a list of available elements into the state machine is also unwanted (and would make automating it needlessly complicated), another way would be to move the logic to the backend library?

When I created this PR, my thoughts were that maybe someone else would also find automating the shuffling of effects & palettes useful. In the end it is not a big deal for me to keep this on my own fork over the holidays, after which it can be ignored until the next holiday season, so I don't mind if you prefer to have this PR closed now. Let me know how you prefer / feel free to close it as you wish.

Reply:
@rytilahti Effect lists for lights are already in the state machine. That is default functionality of Home Assistant.

Reply:
In a service call:

```yaml
- service: wled.effect
  target:
    area_id: living_room
  data:
    effect: "{{ state_attr('light.office_desk_left', 'effect_list') | random }}"
```

Reply:
Ah, indeed, looks like I overlooked that fact. If you don't mind, I'll create a doc PR to add this to wled documentation for others to enjoy?

That still leaves the question about palettes open, any suggestions what's the proper approach for that?

Reply:
For pallets there is not really a good method, however, palettes work by ID, so you could generate a random number between the lowest and highest palette ID. Not as clean, but will do the trick.

There are some possible improvements on the roadmap, that would allow for handling palettes better in Home Assistant (but that requires some core changes first).

Reply:
Okay, I think I keep using my modified integration for now, but I created https://github.com/home-assistant/home-assistant.io/pull/15981 to help anyone else looking to do this & I will close this PR. Thanks for the help!